tests/local_vespa/test_app/services.xml

```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root. -->
+<services version="1.0" xmlns:deploy="vespa" xmlns:preprocess="properties">
+
+    <!--
+        A container cluster handles incoming requests to the application and processes those requests,
+        and their results. The processing to do and the API's to expose can be provides by Vespa
+        or by the application through Java components supplied as part of the application.
+
+        See:
+          - Reference: https://docs.vespa.ai/en/reference/services-container.html
+    -->
+    <container id="default" version="1.0">
+        <!--
+            <document-api> tells the container that it should accept documents for indexing. Through the
+            Document REST API you can PUT new documents, UPDATE existing documents, and DELETE documents
+            already in the cluster.
+
+            Documents sent to the Document REST API will be passed through document processors on the way
+            to the content cluster.
+
+            See:
+             - Reference: https://docs.vespa.ai/en/reference/services-container.html#document-api
+             - Operations: https://docs.vespa.ai/en/document-v1-api-guide.html
+        -->
+        <document-api/>
+
+        <!--
+            <search> tells the container to answers queries and serve results for those queries.
+            Inside the <search /> cluster you can configure chains of "searchers" -
+            Java components processing the query and/or result.
+
+            See:
+             - Reference: https://docs.vespa.ai/en/query-api.html
+             - Searchers: https://docs.vespa.ai/en/searcher-development.html
+        -->
+        <search/>
+
+        <!--
+            <nodes> specifies the nodes that should run this cluster.
+        -->
+        <nodes deploy:environment="dev" count="1">
+            <resources vcpu="2" memory="8Gb" disk="50Gb"/>
+        </nodes>
+        <nodes deploy:environment="perf" count="[1, 4]">
+            <resources vcpu="[2, 4]" memory="[8Gb, 16Gb]" disk="[50Gb, 500Gb]"/>
+        </nodes>
+        <nodes deploy:environment="prod" count="[1, 4]">
+            <resources vcpu="[2, 4]" memory="[8Gb, 16Gb]" disk="[50Gb, 500Gb]"/>
+        </nodes>
```

Comment:
do we want these resource constraints on a local machine? seems a bit high

Reply:
Good question. Its a copy of the prod setup and it'd be nice to keep them identical to each other and I've got plans to add some automation to make sure they stay up to date. Though maybe this is only important for he schema and query profiles...

Reply:
I might be able to sidestep this issue by adding constraints to the docker compose file instead... what do you think?

Reply:
i'm up for some sort of constraints and docker sounds good - i doubt we'll reach them, but good to keep especially disk usage in case no?

Reply:
After looking at this a bit more I'm settling on a dev specific `services.xml` and lowering the vespa resource requirements, i.e. what you suggested initially! 😄 It seems safer than getting the docker compose resource limits right, and there are a few benefits to the services.xml being flexible depending on what gets tested